main.go

```diff
@@ -59,8 +60,33 @@ func run(w io.Writer, commandLine []string) error {
 	return nil
 }
 
-func checkFiles(w io.Writer, rootRule rules.Rule, cfg *config.Config, files []sops.File) error {
+// sarifRun compiles all the results and creates a Sarif run.
+func sarifRun(results []rules.SarifResult) *sarif.Run {
+	run := sarif.NewRunWithInformationURI("sops-check", "sops-check")
+
+	for _, r := range results {
+		run.AddRule(r.RuleID).
+			WithDescription(r.Description)
+
+		run.CreateResultForRule(r.RuleID).
+			WithLevel(strings.ToLower(r.Evaluation)).
+			WithMessage(sarif.NewTextMessage(r.Message)).
+			AddLocation(
+				sarif.NewLocationWithPhysicalLocation(
+					sarif.NewPhysicalLocation().
+						WithArtifactLocation(
+							sarif.NewSimpleArtifactLocation(r.File),
+						),
+				),
+			)
```

Comment:
A couple of temporary variable would be nice here to avoid the nesting. Right now it's a bit hard to see what gets created here ;). I would at least use a `location` variable here.

Reply:
This I didn't change as i think for now its more readable to have it in one place instead of using temp variables. 
We might need to do that once we start also tracking lines where the error happened as the nesting might become too much then